bin/compile

```diff
@@ -153,7 +153,7 @@ install_bins() {
 
   echo "engines.node (package.json):  ${node_engine:-unspecified}"
   echo "engines.npm (package.json):   ${npm_engine:-unspecified (use default)}"
-  if $YARN; then
+  if $YARN || [ -n "$yarn_engine" ]; then
```

Comment:
Do you mind adding a test for this? It can be something similar to this one: https://github.com/heroku/heroku-buildpack-nodejs/blob/master/test/run#L292-L296, but the fixture has a monorepo that just has a yarn engine without the lockfile.